### PR TITLE
Add aten::masked_scatter_

### PR DESCRIPTION
**Pull request by xytintel**
Add aten::masked_scatter_.

### CODE REVIEW

**Comment by xytintel** [comment]
image
image

File: src/ATen/native/xpu/Indexing.cpp [hunk 1]
```diff
@@ -44,4 +43,35 @@ Tensor XPUNativeFunctions::index_select(
...
+
+  c10::MaybeOwned<Tensor> b_mask =
+      expand_inplace(self, mask, "masked_scatter_");
+
+  if (self.numel() == 0) {
+    return self;
+  }
+
+  auto maskPrefixSum = at::empty(self.sizes(), mask.options().dtype(kLong));
+  native::xpu::launch_masked_scatter_kernel(
```
Comment (fengyuan14) [inline src/ATen/native/xpu/Indexing.cpp:71]: To align our style, there should be no launch as prefix.

File: src/ATen/native/xpu/sycl/pstl/PSTLFunctions.h [hunk 2]
```diff
@@ -23,10 +22,10 @@ struct KSScanKernelFunctor : public __SYCL_KER_CONFIG_CONVENTION__ {
     // initialize local_input
     auto cur_init = init_;
     if (scan_type == 1) {
-      local_scan_[local_id] = first_[local_id];
+      local_scan_[local_id] = c10::load(&first_[local_id]);
```
Comment (fengyuan14) [inline src/ATen/native/xpu/sycl/pstl/PSTLFunctions.h:25]: When we should use `c10::load` and when we could use subscript.

File: src/ATen/native/xpu/Indexing.cpp (hunk 1 above)
Comment (xytintel) [reply src/ATen/native/xpu/Indexing.cpp]: Fixed

**Review by fengyuan14 — APPROVED** [review]

**Pull request merged**